src/test/java/server/RouterTest.java

```diff
@@ -0,0 +1,100 @@
+package server;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class RouterTest {
+
+    Response testGETGetResponse = new Response(
+            "protocol",
+            "statusCode",
+            "Route: GET /get",
+            ""
+    );
+
+    Response testHEADGetResponse = new Response(
+            "protocol",
+            "statusCode",
+            "Route: HEAD /get",
+            ""
+    );
+
+    Response testHEADHeadResponse = new Response(
+            "protocol",
+            "statusCode",
+            "Route: HEAD /head",
+            ""
+    );
+
+    Response testNotFoundResponse = new Response(
+            "protocol",
+            "404 Not Found",
+            "Route: not found",
+            ""
+    );
+
+    Map<String, Handler> testRoutes = Map.of(
+            "GET /get", new TestHandler(testGETGetResponse),
+            "HEAD /get", new TestHandler(testHEADGetResponse),
+            "HEAD /head", new TestHandler(testHEADHeadResponse)
+    );
```

Comment:
**suggestion (non-blocking)**: I think you could use the lambda syntax if you wanted to here


```suggestion
            "GET /get", request -> new Response("protocol", "statusCode", "Route: /get", ""),
            // other routes here...
    );
```